From b1c1ad42eb81cd3620b14dcf6bb7962561505b13 Mon Sep 17 00:00:00 2001
From: Dan Williams <dan.j.williams@intel.com>
Date: Thu, 4 Aug 2016 16:53:50 -0700
Subject: dax: unmap/truncate on device shutdown
Git-commit: 9dc1e4927bfabaf654738c9ecca3a4926a0aaeb5
Patch-mainline: v4.9-rc1
References: FATE#321135, FATE#321217, FATE#321256, FATE#321391, FATE#321393

Invalidate all mappings of a device-dax instance when the device is
unregistered.

Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>

---
 drivers/dax/dax.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/dax/dax.c b/drivers/dax/dax.c
index e1a4a36..386b33c 100644
--- a/drivers/dax/dax.c
+++ b/drivers/dax/dax.c
@@ -429,6 +429,7 @@ static void unregister_dax_dev(void *dev)
 	 */
 	dax_dev->alive = false;
 	synchronize_rcu();
+	unmap_mapping_range(dax_dev->inode->i_mapping, 0, 0, 1);
 	cdev_del(cdev);
 	device_unregister(dev);
 }
-- 
1.8.5.6

